bilby/gw/likelihood.py

```diff
@@ -1495,7 +1495,7 @@ def setup_bins(self):
             d_phi_from_start = d_phi - d_phi[0]
             self.number_of_bins = int(d_phi_from_start[-1] // self.epsilon)
             self.bin_freqs[interferometer.name] = np.zeros(self.number_of_bins + 1)
-            self.bin_inds[interferometer.name] = np.zeros(self.number_of_bins + 1)
+            self.bin_inds[interferometer.name] = np.zeros(self.number_of_bins + 1, dtype=np.int)
 
```

Comment:
It's gonna be an integer anyway, right?

Reply:
Yeah, but by default `np.zeros` initializes to a `float` array. So all the elements that one sets also becomes `float`. The `dtype` is explicitly defined to avoid this.